### ISSUE TITLE
NSM: add ability to pin snort via IDS_LB_CPUS

### ISSUE THREAD

**Issue by dougburks** [Security-Onion-Solutions/security-onion#1729]
https://github.com/Security-Onion-Solutions/securityonion-nsmnow-admin-scripts/pull/28

**Comment by dougburks** [comment]
Copied from https://github.com/Security-Onion-Solutions/securityonion-nsmnow-admin-scripts/pull/28:

This may be a starting point for the ability to pin snort processes to specific CPUs by adding a line to the /etc/nsm/<sensor>/sensor.conf file like:

> IDS_LB_CPUS=1,3,5,7
 
As specified, the first four snort processes would be pinned to the first four odd-numbered CPU cores.  It validates the input as a number before using it, so if there are more than the specified number (eg 5), any processes without a CPU listed would have the default CPU affinity.

This is admittedly a little hackish, as the CPU for taskset should probably be specified as a seventh parameter to process_start() instead of specifying both taskset and the command to run in the first (CMD) parameter.

To test, I recommend first moving all processes to a specific set of CPUs using systemd's CPUAffinity setting in /etc/systemd/system.conf and specifying values for --cpuset-cpus in the docker containers using the *_OPTIONS values in /etc/nsm/securityonion.conf.  The htop utility can be useful for verifying that the CPUs not given there are idle.  Then, you can test this setting and verify that the processes are moved to the CPUs idled earlier.

**Comment by dougburks** [comment]
You can also verify proper pinning using `taskset -p PID` where `PID` is the actual process ID of the Snort process you are checking.  The default affinity mask is `f`, so if pinning is set properly you should see something other than that.

**Comment by bryant-treacle** [comment]
Everything tested good.  I used the following command to check the cpu affinity.  taskset -cp PID.  It gave me the numerical list of processors instead of a bitmask.  Just a little easier to read the results.  

Tested 
IDS_LB_PROCS=3
IDS_LB_CPUS=1,3,5
Here each Snort process was pinned to an individual CPU

Tested 
IDS_LB_PROCS=3
IDS_LB_CPUS=1,3
The Snort-3 process took the default Affinity set in /etc/systemd/system.conf

The settings persisted through a reboot. 

Ran so-test - IDS alerts were generated with no errors in logs or dropped packets.
